herddb-core/src/main/java/herddb/core/TableSpaceManager.java

```diff
@@ -2068,7 +2068,13 @@ public void accept(LogSequenceNumber t, LogEntry u) {
                 throw new RuntimeException("System was requested to stop, aborting recovery at " + t);
             }
             try {
-                apply(new CommitLogResult(t, false, true), u, true);
+                if (u.type != LogEntryType.TABLE_CONSISTENCY_CHECK) {
+                    apply(new CommitLogResult(t, false, true), u, true);
```

Comment:
why don't you push this check into "apply" ?

Reply:
right